### PR TITLE
Np dev

### PR DESCRIPTION
**Pull request by NiklasPfister**
Restructured the documentation and added some place holder files.

@svbrodersen: Please double check everything looks fine and then merge.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,65 @@
...
 
-**adaXT** is a Python module for tree-based regression and classification that is fast, adaptable and extendable and aims to provide researchers a more flexible workflow when building tree-based models.
+**adaXT** is a Python module for tree-based machine-learning
+algorithms that is fast, adaptable and extendable and aims to provide
+researchers a more flexible workflow when developing tree-based
+models.
 
-### Installation
-adaXT is available at [pypi](https://pypi.org/) and can be installed by running ```pip install adaXT```.
+It is distributed under the [3-Clause BSD license](LICENSE).
```
Comment (svbrodersen) [inline README.md:8]:
```suggestion
It is distributed under the [3-Clause BSD license](https://github.com/NiklasPfister/adaXT/blob/main/LICENSE).
```
As the license doesn't have it's own page, you can't use the default referencing and you have
to leave a link to the site itself.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,65 @@
-## adaXT - Fast adaptable and extendable trees for research
+## Fast Adaptable and Extendable Trees for Research
 
-**adaXT** is a Python module for tree-based regression and classification that is fast, adaptable and extendable and aims to provide researchers a more flexible workflow when building tree-based models.
+**adaXT** is a Python module for tree-based machine-learning
+algorithms that is fast, adaptable and extendable and aims to provide
```
Comment (svbrodersen) [inline README.md:4]:
```suggestion
algorithms that is fast, adaptable and extendable and aims to provide
```

File: README.md [hunk 3]
```diff
@@ -1,6 +1,65 @@
...
+#### Adaptable and extendable
+
+At the heart of any tree-based algorithm is a decision tree that can
+be fitted on data and then used to perform some version of
+prediction. adaXT has therefore been designed with a modular decision
+tree implementation that takes four input components:
+
+- Criterion class: Used during fitting to determine splits.
+
+- LeafBuilder class: Used during fitting to specify what is saved in
```
Comment (svbrodersen) [inline README.md:39]: This should not be added until the given Prediction and LeafBuilder are on the main branch.
And preferably even in a new release.

File: README.md [hunk 4]
```diff
@@ -1,6 +1,65 @@
...
+  the leaf nodes.
+
+- Splitter class: Used during fitting to perform the splits.
+
+- Prediction class: Used after fitting to make predictions.
+
+By specifying these three components a range of different tree
```
Comment (svbrodersen) [inline README.md:46]:
```suggestion
By specifying these four components a range of different tree
```
I know that the splitter is the same for all, but stating four inputs, and then stating 
three components seems odd. If you want to keep it as three, perhaps add the specific names
e.g. Predict, Criteria and LeafBuilder.

File: README.md [hunk 5]
```diff
@@ -1,6 +1,65 @@
...
+this modular structure, all other operations are kept as vanilla as
+possible allowing users to easily change parts of the code (e.g., the
+splitting procedure).
+
+
+#### Speed
+
+As tree-based algorithms involve evaluating expensive loops over the
+dataset, it is important that these computations are implemented in a
+compiled language. adaXT implements all computationally expensive
```
Comment (svbrodersen) [inline README.md:58]: The same about capitalization of AdaXT.

File: README.md [hunk 6]
```diff
@@ -1,6 +1,65 @@
...
+operations in [Cython](https://cython.org/). This results in speeds
+similar (although a few factors slower) than the corresponding
+[scikit-learn](https://scikit-learn.org) implementations. However, due
+to its modular structure and the avoidence of technical speed-ups
```
Comment (svbrodersen) [inline README.md:62]:
```suggestion
to its modular structure and the avoidance of technical speed-ups
```

File: README.md [hunk 7]
```diff
@@ -1,6 +1,65 @@
...
+- LeafBuilder class: Used during fitting to specify what is saved in
+  the leaf nodes.
+
+- Splitter class: Used during fitting to perform the splits.
+
+- Prediction class: Used after fitting to make predictions.
+
+By specifying these three components a range of different tree
+algorithms can be created, e.g., regression trees, classification
+trees, quanitle regression trees and survial trees. Additionally to
```
Comment (svbrodersen) [inline README.md:48]:
```suggestion
trees, quantile regression trees and survival trees. Additionally to
```

File: README.md [hunk 8]
```diff
@@ -1,6 +1,65 @@
...
+The goal of adaXT is to provide a flexible and unified code-base for
+various tree-based algorithms that strikes a balance between speed and
+ease with which the code can be adapted and extended. It intends to
+provide researchers a simple toolkit for prototyping new tree-based
+algorithms.
+
+adaXT aims to provide an intuitive user experience that is similar to
+the [scikit-learn](https://scikit-learn.org) implementations of
+decision trees both in terms model-based syntax and
+hyperparameter. Under the hood, however, adaXT stikes a different
```
Comment (svbrodersen) [inline README.md:27]:
```suggestion
hyperparameter. Under the hood, however, adaXT strikes a different
```

File: docs/api_docs/DecisionTree.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+- [Prediction]()
+
+Instead of the user specifying all three components individually, it
+is also possible to only specify the ``tree_type``, which then
+internally selects the corresponding default components for several
+established tree-algorithms.
+
+For more advanced modifications, it might be necessary to change the
+splitting itself. This can be done by passing a custom [Splitter
+class](../splitter/splitter.md).
```
Comment (svbrodersen) [inline docs/api_docs/DecisionTree.md:18]:
```suggestion
class](Splitter.md).
```
You want a link to the given Splitter markdown, which is in the same folder as the 
current markdown file.

File: docs/api_docs/LeafBuilder.md [hunk 10]
```diff
@@ -0,0 +1,3 @@
+# LeafBuilder Class
+
+TODO
```
Comment (svbrodersen) [inline docs/api_docs/LeafBuilder.md:3]:
```suggestion
TODO
```

I would argue that the PR should go to the Development branch,
as long as there still exists TODO's. The main branch should be fully complete for a given version.

File: docs/api_docs/Prediction.md [hunk 11]
```diff
@@ -0,0 +1,3 @@
+# Prediction Class
+
+TODO
```
Comment (svbrodersen) [inline docs/api_docs/Prediction.md:3]: Same for the Branch here.

File: docs/api_docs/Splitter.md [hunk 12]
```diff
@@ -0,0 +1,6 @@
+# Splitter Class
```
Comment (svbrodersen) [inline docs/api_docs/Splitter.md:1]: And here.

File: src/adaXT/criteria/criteria.pyi [hunk 13]
```diff
@@ -1,56 +1,71 @@
...
+    - \Big(\tfrac{1}{|I|}\sum_{i\in I} Y[i]\Big)^2
     $$
     """
     pass
 
 
 class Linear_regression(Criteria):
     r"""
-    Calculates the impurity given by
+    Linear regression based criteria, which adapts the Squared_error
```
Comment (svbrodersen) [inline src/adaXT/criteria/criteria.pyi:58]: I wouldn't say that the Linear Regression adapts the Squared_error criterion.

File: src/adaXT/decision_tree/DecisionTree.pyi [hunk 14]
```diff
@@ -71,20 +71,20 @@ class DecisionTree:
         sample_weight: np.ndarray | None = None,
     ) -> None:
         """
-        Function used to fit the data on the tree using the DepthTreeBuilder
+        Bild the decision tree from the training data (X, y).
```
Comment (svbrodersen) [inline src/adaXT/decision_tree/DecisionTree.pyi:74]:
```suggestion
        Build the decision tree from the training data (X, y).
```

**Review by svbrodersen — CHANGES_REQUESTED** [review]
Overall looks fine, however I have to checkout to the branch and do some testing.

I am pretty sure the majority of the links are broken after the restructering, as the way linking works 
is through the structure of the markdown files. 

Further, added a few spell checks and will try and look through the linking problems and fix any I find.

File: README.md (hunk 1 above)
Comment (NiklasPfister) [reply README.md]: Why is the full url not needed in the current version?

File: README.md (hunk 3 above)
Comment (NiklasPfister) [reply README.md:39]: I would like to have this updated together with your proposed change that adds the leafbuilder and prediction class.

File: docs/api_docs/LeafBuilder.md (hunk 10 above)
Comment (NiklasPfister) [reply docs/api_docs/LeafBuilder.md:3]: Yes I actually wanted to merge it to the development branch...

File: src/adaXT/criteria/criteria.pyi (hunk 13 above)
Comment (NiklasPfister) [reply src/adaXT/criteria/criteria.pyi:58]: I think adapts is fine. While its a different criterion it uses squared error in all but the first coordinate and the first coordinate can be seen as an adaption in my opinion.

**Comment by NiklasPfister** [comment]
Thanks for checking! I now changed the base to "development", thats also what I originally intended.

File: README.md (hunk 1 above)
Comment (svbrodersen) [reply README.md]: Not sure I follow. The current link doesn't work? If you do mkdocs serve, it will send you 
to a 404 page not found error.

**Pull request merged**